### PR TITLE
scan all ports

### PR DESCRIPTION
**Pull request by progers-tech**
mo' ports

### CODE REVIEW

File: infrastructure/elastic_resources/searches/non_standard_ports.search.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+    "attributes": {
+        "title": "${object_title}",
+        "description": "A search returning all IP addresses where a port is open that isn't 80, 443, 22, and some mail ones",
```
Comment (blackmamo) [inline infrastructure/elastic_resources/searches/non_standard_ports.search.json]: Do we want to hard code these or use sliders?

**Review by blackmamo — CHANGES_REQUESTED** [review]
I think that this is probably updated now anyway since you showed me this with sliders, but can we not hardcode 443, 80, 22?. CAn you also check that task execution and shared code references are updated to latest on master

File: infrastructure/elastic_resources/dashboards.tf [hunk 2]
```diff
@@ -82,10 +82,67 @@ module "moj_dashboard_3" {
...
   object_title = "SSL Expiry"
   es_domain    = data.aws_ssm_parameter.es_domain.value
 }
 
+
+module "moj_dash_ai_open_ports" {
+  // two slashes are intentional: https://www.terraform.io/docs/modules/sources.html#modules-in-package-sub-directories
+  source = "github.com/ministryofjustice/securityanalytics-analyticsplatform//infrastructure/kibana_saved_object"
+
+  // It is sometimes useful for the developers of the project to use a local version of the task
```
Comment (blackmamo) [inline infrastructure/elastic_resources/dashboards.tf:99]: # not // please, this applies to all chages

File: infrastructure/docker_image/Dockerfile.template [hunk 3]
```diff
@@ -11,4 +11,4 @@ COPY task_script.sh .
 ENV NMAP_TARGET_STRING=127.0.0.1
 ENV RESULTS_BUCKET=${results_bucket_arn}
 
-ENTRYPOINT [ "sh", "task_script.sh", "--resolve-all", "--script", "ssl-enum-ciphers,vulners,ssl-cert", "-A", "-v" ]
\ No newline at end of file
+ENTRYPOINT [ "sh", "task_script.sh", "--resolve-all", "--script", "ssl-enum-ciphers,vulners,ssl-cert", "-A", "-v", "-sT", "-p0-65535", "-T4"]
```
Comment (blackmamo) [inline infrastructure/docker_image/Dockerfile.template:14]: We now have way too many arguments to not comment on them. We should explain exactly what options we are passing and why

File: results_parser/ssl-cert.py [hunk 4]
```diff
@@ -1,10 +1,16 @@
...
+
+
 def process_script(script, results_context):
     results = {}
     for table in script.table:
         if table["key"] == "validity":
             results["validity"] = {}
             for elem in table.elem:
                 results["validity"][elem["key"]] = elem.cdata
+                if elem["key"] == "notAfter":
```
Comment (blackmamo) [inline results_parser/ssl-cert.py:11]: This method is a bit of a monster now. PLease consider splitting it up as a way to add more names and extend understanding

**Review by blackmamo — CHANGES_REQUESTED** [review]

**Comment by progers-tech** [comment]
in the other PR

**Pull request closed without merging**